### PR TITLE
fix: migrate delete personal card api to platform

### PR DESCRIPTION
**Pull request by sumrender**
## Clickup
https://app.clickup.com/t/86cwm1x50

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

## Summary by CodeRabbit

- **New Features**
	- Introduced a feature flag management system via `LaunchDarklyService` for account deletion.
	- Enhanced account deletion method to support both public and platform APIs.
  
- **Bug Fixes**
	- Improved type safety for event handling in the `presentPopover` method.

- **Tests**
	- Added new test cases for the `deleteAccount()` method to ensure proper functionality with both APIs.
	- Updated test suite for `BankAccountCardComponent` to reflect changes in method signatures and dependencies.

- **Documentation**
	- Clarified asynchronous return types for several methods in the `BankAccountCardComponent`.

### CODE REVIEW

File: src/app/core/models/platform/platform-api-response.model.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export interface PlatformApiResponse<T> {
-  count: number;
-  offset: number;
+  count?: number;
```
Comment (hlkavya0213) [inline src/app/core/models/platform/platform-api-response.model.ts:2]: in platform API response count & offset will always be there right ?

File: src/app/core/services/personal-cards.service.spec.ts [hunk 2]
```diff
@@ -134,6 +135,40 @@ describe('PersonalCardsService', () => {
     });
   });
 
+  describe('deleteAccount()', () => {
+    it('should delete personal card when using public api', (done) => {
```
Comment (hlkavya0213) [inline src/app/core/services/personal-cards.service.spec.ts:139]: why are we still keeping the public API?

File: src/app/core/services/personal-cards.service.ts [hunk 3]
```diff
@@ -124,7 +124,21 @@ export class PersonalCardsService {
...
+      data: {
+        id: accountId,
+      },
+    };
+    return this.spenderPlatformV1ApiService
+      .post<PlatformApiResponse<PersonalCardPlatform>>('/personal_cards/delete', payload)
+      .pipe(map((res) => res.data));
+  }
+
+  deleteAccount(accountId: string, usePlatformApi: boolean): Observable<PersonalCard | PersonalCardPlatform> {
```
Comment (hlkavya0213) [inline src/app/core/services/personal-cards.service.ts:138]: why do we still need this?

File: src/app/core/services/personal-cards.service.ts [hunk 4]
```diff
@@ -124,7 +124,21 @@ export class PersonalCardsService {
...
-  deleteAccount(accountId: string): Observable<PersonalCard> {
+  deleteAccountPlatform(accountId: string): Observable<PersonalCardPlatform> {
+    const payload = {
+      data: {
+        id: accountId,
+      },
+    };
+    return this.spenderPlatformV1ApiService
+      .post<PlatformApiResponse<PersonalCardPlatform>>('/personal_cards/delete', payload)
+      .pipe(map((res) => res.data));
```
Comment (hlkavya0213) [inline src/app/core/services/personal-cards.service.ts:135]: better name for res

File: src/app/shared/components/bank-account-cards/bank-account-card/bank-account-card.component.spec.ts [hunk 5]
```diff
@@ -13,8 +13,9 @@ import { ToastMessageComponent } from '../../toast-message/toast-message.compone
 import { PopupAlertComponent } from '../../popup-alert/popup-alert.component';
 import { DeleteButtonComponent } from './delete-button/delete-button-component';
 import { click, getElementBySelector, getTextContent } from 'src/app/core/dom-helpers';
+import { LaunchDarklyService } from 'src/app/core/services/launch-darkly.service';
 
-describe('BankAccountCardComponent', () => {
+fdescribe('BankAccountCardComponent', () => {
```
Comment (hlkavya0213) [inline src/app/shared/components/bank-account-cards/bank-account-card/bank-account-card.component.spec.ts:18]: why are we changing to fdescribe?

**Review by hlkavya0213 — CHANGES_REQUESTED** [review]
some comments

File: src/app/shared/components/bank-account-cards/bank-account-card/bank-account-card.component.spec.ts (hunk 5 above)
Comment (sumrender) [reply src/app/shared/components/bank-account-cards/bank-account-card/bank-account-card.component.spec.ts:18]: Resolved

File: src/app/core/models/platform/platform-api-response.model.ts (hunk 1 above)
Comment (hlkavya0213) [reply src/app/core/models/platform/platform-api-response.model.ts:2]: check this

**Review by hlkavya0213 — APPROVED** [review]

File: src/app/core/models/platform/platform-api-response.model.ts (hunk 1 above)
Comment (sumrender) [reply src/app/core/models/platform/platform-api-response.model.ts:2]: This is only when the response is an array. Here in delete personal card api, I am receiving only the data property in the response.

**Pull request merged**